pkg/odo/cli/component/common_link.go

```diff
@@ -303,7 +321,10 @@ func (o *commonLinkOptions) validateForOperator() (err error) {
 			},
 		}
 	}
+
+	service.Id = &o.serviceName // Id field is helpful if user wants to inject mappings (custom binding data)
 	o.serviceBinding.Spec.Services = []servicebinding.Service{service}
+	o.serviceBinding.Spec.Mappings = o.MappingsMap
```

Comment:
Could you make some refactoring to have a function `f` :

```
o.serviceBinding = f(arg1, erg2, ...)
```

This way:
- we can more easily see won what `o.serviceBinding` depends (arg1, erg2, etc)
- f will be more easily testable